compiler/src/language_server/driver.re

```diff
@@ -41,7 +41,7 @@ let process = msg => {
     Code_file.DidChange.process(~uri, ~compiled_code, ~documents, params);
     Reading;
   | Formatting(id, params) when is_initialized^ =>
-    Formatting.process(~id, ~compiled_code, ~documents, params);
+    Formatting.process(~id, ~documents, params);
```

Comment:
Why did you remove `~compiled_code`? It is passed to every "behavior" to keep them uniform, even if it is unused.

Reply:
Oooh, that's very opposite to how I work, I only ever pass in what a function needs so its dependencies are clear, but I'll put it back.

Reply:
@marcusroberts I'll explain a little more: When writing this code, I envisioned that each "action" the LSP could handle would be a specialization to a module functor that provides an implementation. I didn't implement it that way to see if this pattern would work (and for the sake of time/effort), but for that to work in the future, we'll need to keep the module signatures the same.